gui/src/store/settings.ts

```diff
@@ -0,0 +1,39 @@
+import { appWindow } from "@tauri-apps/api/window";
+import { Action } from "kbar";
+import { StateCreator, create } from "zustand";
+import { subscribeWithSelector } from "zustand/middleware";
+
+interface State {
+  show: boolean;
+  actions: Action[];
+}
+
+interface Setters {
+  open: () => unknown;
+  close: () => unknown;
+}
```

Comment:
when I saw a settings store, I expected something very different, equivalent to the other ones: to sync data from the settings that exist on the backend, and replace all `invoke` calls around the app

instead it's a store about the "settings window", not the settings themselves. is there a more clear name we can use?

Reply:
I understand, but I don't have great ideas. We do need state management for temporary things. What about calling it "window" store? This could evolve to control more pieces of information related to windows.

Reply:
> This could evolve to control more pieces of information related to windows.

is that the actual plan, or just a random idea?

if we don't really know how this will evolve, and what other data will be here, I'd rather just pick a more explicit name and rename it again later once we figure that out

`settings` to me is confusing because, from seeing all the other stores right now (balances, contracts, wallets, theme). They all store the thing described by the name. this one doesn't store the settings, but stores the "state of the setings window"

so something like "settingsWindow" works for me

"window" is also fine except that: 1) it's a reserved keyword already, and 2) means `open/close` need to be renamed to `openSettings / closedSettings` (which is fine to me)